nixos/modules/services/web-servers/caddy/default.nix

```
@@ -41,6 +41,14 @@ let
     in
       "${if pkgs.stdenv.buildPlatform == pkgs.stdenv.hostPlatform then Caddyfile-formatted else Caddyfile}/Caddyfile";
 
+  adminDisabled = lib.fileContents (pkgs.runCommand "caddy-config-adapted" {} ''
+    ${cfg.package}/bin/caddy adapt --config ${configFile} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"} | ${pkgs.jq}/bin/jq .admin.disabled > $out
```

```suggestion
    ${cfg.package}/bin/caddy adapt --config ${cfg.configFile} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"} | ${pkgs.jq}/bin/jq .admin.disabled > $out
```

If checking if admin is disabled is done here (at eval time), the this change should be made.  `configFile` is generated by this module and is the default value of `cfg.configFile`.  However, users can also specified their own `cfg.configFile`.

If checking if admin is disabled is done in `ExecStartPre` as I proposed before (at run time), `configFile` should changed to `configPath`.